packages/medusa/src/instrumentation/index.ts

```diff
@@ -117,6 +117,46 @@ export function instrumentHttpLayer() {
   })
 }
 
+/**
+ * Instrumenting the queries made using the remote query
+ */
+export function instrumentRemoteQuery() {
+  const QueryTracer = new Tracer("@medusajs/query", "2.0.0")
+
+  Query.traceGraphQuery = async function (queryFn, queryOptions) {
+    return await QueryTracer.trace(
+      `query.graph: ${queryOptions.entryPoint}`,
+      async (span) => {
+        span.setAttributes({
+          "query.fields": queryOptions.fields,
+        })
+        return await queryFn().finally(() => span.end())
+      }
+    )
+  }
+
+  Query.traceRemoteQuery = async function (queryFn, queryOptions) {
+    const traceIdentifier =
```

Comment:
todo: here you are manipulating the options like it was the config, maybe we need both as mentioned above :) also, the config is supposed to be unwrapped when it reaches the remote query